### ISSUE TITLE
[REQUEST] Does sequence parallel need all-reduce for gradients before optimizer.step()?

### ISSUE THREAD

**Issue by zxmeng98** [microsoft/DeepSpeed#4548]
Hi, 
I am reading the implementation code of sequence parallelism recently and didn't find the all-reduce of gradients after backward finishes. Because the sequence is split to sub-sequences before fed into network, I wonder should there be an all-reduce for gradients before optimizer.step()? Similar to data parallelism?

In https://github.com/microsoft/Megatron-DeepSpeed/blob/main/megatron/training.py#L638:
![dcf7a9afd4e5e0fdf6b759a6f490451](https://github.com/microsoft/DeepSpeed/assets/96046956/db567cf3-8bbf-4afe-a79e-33a5f218fb30)
